katdal/sensordata.py

```diff
@@ -389,15 +389,15 @@ def dummy_sensor_data(name, value=None, dtype=np.float64, timestamp=0.0):
 
     """
     if value is None:
-        if np.issubdtype(dtype, np.float):
-            value = np.nan
-        elif np.issubdtype(dtype, np.int):
-            value = -1
-        elif np.issubdtype(dtype, np.str):
+        if np.issubdtype(dtype, np.floating):
+            value = np.dtype(dtype).type(np.nan)
+        elif np.issubdtype(dtype, np.integer):
+            value = np.dtype(dtype).type(-1)
```

Comment:
Are unsigned types used anywhere?

Reply:
Don't think so - the default value is pretty arbitrary in this case anyway.